### ISSUE TITLE
Incorrect type. Expected "string"

### ISSUE THREAD

**Issue by carlbuchmann** [redhat-developer/vscode-yaml#96]
When writing ansible playbook, I get many reported problems where keys contain boolean values or integers... This seems that have started since the last update 4-5 days ago...
example:
![image](https://user-images.githubusercontent.com/20460062/41804052-47cee9aa-765e-11e8-8519-dc39a44d4c9a.png)

Reproduced on both my windows and Linux development workstations with the latest version of vscode: 1.24.1

**Comment by gorkem** [comment]
Can be related to https://github.com/redhat-developer/vscode-yaml/issues/93

**Comment by cmendible** [comment]
Hi guys, same here!  It makes my yaml files almost unreadable. Seems to happen with hashes with the latest update.

![expectstring](https://user-images.githubusercontent.com/266546/41921247-b86820bc-7962-11e8-9289-68bd531e254a.PNG)

**Comment by JPinkney** [comment]
It has nothing to do with the latest update. It has to do [ansible files being added to schemastore](https://github.com/SchemaStore/schemastore/commit/8af13cd3afa7ca67a1c246b38e19602c7e242783). This extension pulls from schema store to automatically configure specific well used files. Please file an issue over there and then it will be updated.

**Comment by cmendible** [comment]
Hmmm that makes sense! I'll check with them...

Thanks!

**Comment by cmendible** [comment]
I've created the following pull request to fix my issue: https://github.com/SchemaStore/schemastore/pull/480

**Comment by carlbuchmann** [comment]
it looks like the schema files are auto-generated from ansible documentation, so I think it's best to go and fix the source on the ansible repo... it will break again when 2.6 is released!
"Auto-Generated JSON Schema for Ansible-stable 2.5 (https://github.com/shaded-enmity/ansible-schema-generator" - generate JSON schema for language servers from Ansible module documentation in an Ansible repository checkout target_path

**Comment by cmendible** [comment]
Ok so I can confirm that this issue has nothing to do with this bug. 

We've modified the schemas published in the schema store (https://github.com/SchemaStore/schemastore/pull/480)  as quick fix but both **ansible** documentation (https://github.com/ansible/ansible/pull/42033) and the **ansible-schema-generator** code (https://github.com/shaded-enmity/ansible-schema-generator/issues/1) must be modified.

**Comment by carlbuchmann** [comment]
@cmendible did the quick fix work for you? 
Since there are 5 schemas published for ansible, which one will be used by this YAML extension... does it detect the ansible version of the local host it is running from or is there a way to manually specify? I'm asking because typically support various major versions currently 2.4.x and 2.5.x

**Comment by cmendible** [comment]
Hi @carlbuchmann, it did! I don't have ansible installed (I use cloud shell) so it's not detected from local installation... The extension is using the latest ansible schema from:  http://json.schemastore.org/ansible-stable-2.5

**Comment by shaded-enmity** [comment]
@carlbuchmann There's no way to detect Ansible version at the host, as the schema application logic uses simple path matching. The correct way to approach this IMHO would be to allow the user to select which version of schema to use (if there are more) via the UI and defaulting to the latest. Similar to how I can choose Python2 or Python3.

**Comment by shaded-enmity** [comment]
BTW. I finally realized what is going on in here with the bool/string thing - modules from `ansible` core are processed first, but those modules contain ill-formed documentation which is disjointed from the non-core modules (note that all docs point to the non-core variants e.g https://docs.ansible.com/ansible/latest/modules/service_module.html).
I've pushed a commit that fixes that and I'll try to regenerate the schemas today in the evening.
